### PR TITLE
mds: support alternate names for snapshots

### PR DESCRIPTION
**Pull request by luis-henrix**
This PR adds support for alternate_names in snapshots.  It basically stores the alternate_name already received in SnapInfo and later, when doing an LSSNAP, returns it in the Lease.

Signed-off-by: Luís Henriques <lhenriques@suse.de>

### CODE REVIEW

**Review by lxbsz — COMMENTED** [review]
@luis-henrix  This looks good to me. Please remove the RFC. Thanks.

**Comment by luis-henrix** [comment]
Thanks @lxbsz I've changed the subject.

**Review by jtlayton — APPROVED** [review]

**Review by lxbsz — APPROVED** [review]

**Review by ddiss — APPROVED** [review]
LGTM. Might be worth adding an `alternate_name` parameter to the `LeaseStat` constructor, but it's fine as is.

**Comment by luis-henrix** [comment]
> LGTM. Might be worth adding an `alternate_name` parameter to the `LeaseStat` constructor, but it's fine as is.

Ah, good point!  That's probably more a better C++ style indeed.  I'll re-push my branch with that change in a bit.  That's for the review, @ddiss

**Comment by vshankar** [comment]
> **Edit**: After chatting with David, I've reconsidered this change and, unless someone else feels like it would really make sense, I'm **not** modifying `LeaseStat` to set the alternate_name through the constructor.

I'm ok with that.

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by ddiss** [comment]
It's unclear why this never got merged - it has two official approvals. @vshankar did something go wrong during testing?

**Comment by vshankar** [comment]
> It's unclear why this never got merged - it has two official approvals. @vshankar did something go wrong during testing?

I was never able to put this to test since we had lots of infra issues with teuthology. Things are back working now, but we have a BIG backlog of PRs and an overloaded test queue.

It's on my list for testing but it'll take some more time to get merged.

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by luis-henrix** [comment]
@vshankar Any idea when this could be merged?  I was under the impression that the kernel client side would be pushed into mainline during the current (6.3) merge window, and it would be great to have this PR merged.  (And maybe also #46357, but that's not really related with fscrypt).

**Comment by vshankar** [comment]
> @vshankar Any idea when this could be merged? I was under the impression that the kernel client side would be pushed into mainline during the current (6.3) merge window, and it would be great to have this PR merged. (And maybe also #46357, but that's not really related with fscrypt).

I am struggling a bit to get PRs testing in teuthology. The lab is kind of stable now, but we have a good amount of tests in queue: https://pulpito.ceph.com/queue/. At this time of writing this change in queued for running (wip-vshankar-testing-* branches). I hope the tests run without any infra issues cropping up. If things go well, this change should get merged next week.

**Comment by luis-henrix** [comment]
> I am struggling a bit to get PRs testing in teuthology. The lab is kind of
> stable now, but we have a good amount of tests in queue:
> https://pulpito.ceph.com/queue/. At this time of writing this change in queued
> for running (wip-vshankar-testing-* branches). I hope the tests run without any
> infra issues cropping up. If things go well, this change should get merged next
> week.

Thanks for the update, Venky.  I just found that PR #45073 hasn't been
merged either.  But I don't know how to remove the 'stale' label (without
pushing something into the branch again, of course).

**Comment by lxbsz** [comment]
jenkins retest this please

**Comment by vshankar** [comment]
> I am struggling a bit to get PRs testing in teuthology. The lab is kind of stable now, but we have a good amount of tests in queue: https://pulpito.ceph.com/queue/. At this time of writing this change in queued for running (wip-vshankar-testing-* branches). I hope the tests run without any infra issues cropping up. If things go well, this change should get merged next week.
> Thanks for the update, Venky. I just found that PR #45073 hasn't been merged either. But I don't know how to remove the 'stale' label (without pushing something into the branch again, of course).

#45073 is in my test backlog. Test runs are progressing somewhat, so that PR should be picked up soon. I'll keep you updated.

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#09-Mar-2023

**Comment by vshankar** [comment]
@luis-henrix @lxbsz This requires backport to reef, yes?

**Comment by luis-henrix** [comment]
Venky Shankar ***@***.***> writes:

> @luis-henrix @lxbsz This requires backport to reef, yes?

I admit I haven't been following all the fscrypt-related PRs and where
they've been backported to... but this will definitely be required in any
release where fscrypt is supposed to work.

Cheers,
-- 
Luís

**Comment by vshankar** [comment]
> Venky Shankar ***@***.***> writes:
> @luis-henrix @lxbsz This requires backport to reef, yes?
> I admit I haven't been following all the fscrypt-related PRs and where they've been backported to... but this will definitely be required in any release where fscrypt is supposed to work. Cheers, -- Luís

Which mean q/r. @lxbsz

**Comment by lxbsz** [comment]
> > Venky Shankar _**@**_.***> writes:
> > @luis-henrix @lxbsz This requires backport to reef, yes?
> > I admit I haven't been following all the fscrypt-related PRs and where they've been backported to... but this will definitely be required in any release where fscrypt is supposed to work. Cheers, -- Luís
> 
> Which means q/r. @lxbsz

Yeah, agree with Luis, this need to backport the reef. I will do that after Luis' another PR gets merged.

**Comment by lxbsz** [comment]
> > > > Venky Shankar _**@**_.***> writes:
> > > > @luis-henrix @lxbsz This requires backport to reef, yes?
> > > > I admit I haven't been following all the fscrypt-related PRs and where they've been backported to... but this will definitely be required in any release where fscrypt is supposed to work. Cheers, -- Luís
> > > 
> > > 
> > > Which means q/r. @lxbsz
> > 
> > 
> > Yeah, agree with Luis, this need to backport the reef. I will do that after Luis' another PR gets merged.
> 
> Ah - only to reef and not quincy, right?

It seems some patches have already been in quincy, if so I think we need to backport to quincy too.

I need to investigate this when backporting the fscrypt PRs.

**Pull request merged**
